app/src/main/cpp/skyline/services/glue/ITimeZoneService.cpp

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include <os.h>
+#include <kernel/types/KProcess.h>
+#include <services/timesrv/common.h>
+#include <services/timesrv/results.h>
+#include <services/timesrv/ITimeZoneService.h>
+#include <services/timesrv/core.h>
+#include "ITimeZoneService.h"
+
+namespace skyline::service::glue {
+    ITimeZoneService::ITimeZoneService(const DeviceState &state, ServiceManager &manager, std::shared_ptr<timesrv::ITimeZoneService> core, timesrv::core::TimeServiceObject &timesrvCore, bool writeable) : BaseService(state, manager), core(std::move(core)), timesrvCore(timesrvCore), locationNameUpdateEvent(std::make_shared<kernel::type::KEvent>(state, false)), writeable(writeable) {}
+
+    Result ITimeZoneService::GetDeviceLocationName(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        return core->GetDeviceLocationName(session, request, response);
+    }
+
+    Result ITimeZoneService::SetDeviceLocationName(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        if (!writeable)
+            return timesrv::result::PermissionDenied;
+
+        auto locationName{span(request.Pop<timesrv::LocationName>()).as_string(true)};
+
+        // TODO Checked
+        auto timeZoneBinaryFile{state.os->assetFileSystem->OpenFile(fmt::format("tzdata/zoneinfo/{}", locationName))};
+        std::vector<u8> timeZoneBinaryBuffer(timeZoneBinaryFile->size);
+        timeZoneBinaryFile->Read(timeZoneBinaryBuffer);
+        auto ret{core->SetDeviceLocationNameWithTimeZoneBinary(span(locationName).as_string(true), timeZoneBinaryBuffer)};
+        if (ret)
+            return ret;
```

Comment:
Disagree, Ret is very common in other projects and easier to type

Reply:
The issue is moreso that `ret` is stupid as it doesn't at all imply what the contents of it are which is what variable names are for rather than what the variable will be used for

Reply:
Neither does result?

Reply:
Isn't the return value's type `Result` and therefore contains a `result` value?

Reply:
Yes, just the same as ret contains a return value

Reply:
No, `ret` describes what's done with a variable rather than it's contents. The variable's contents are not `return` but a result instead.

Reply:
Smh you're wrong but I concede

Reply:
Music to my ears :relaxed: